### ISSUE TITLE
Add a "ZFS Intro for Ansible-NAS users" page

### ISSUE THREAD

**Issue by scotws** [davestephens/ansible-nas#73]
Ansible-NAS currently assumes that the user will know how to setup and configure ZFS, including snapshot management. It would be nice to at least point out that a program such as Sanoid (https://github.com/jimsalterjrs/sanoid/) can be used to automate the snapshot part of the NAS. 

(Possibly this could be included in a "default ZFS scenario" for people not so familiar with these things as a reference where to start: Install root on "normal" ext4 file system, create pool assuming two mirrored hard drives, set up datasets for TV, Samba shares, movies, etc, set up snapshots of said datasets.)

**Comment by davestephens** [comment]
Never heard of it but this looks awesome, will check it out!

**Comment by scotws** [comment]
As in discussion for #69 , include the Sanoid stuff in an intro page documenting the basics of ZFS, including suggestions on what parameters to use. Might be worth waiting for ZoL 0.8 to be released for this because it includes cool stuff like TRIM support and native at-rest encryption. Should include example for snapshots etc. for most common use case.

**Comment by scotws** [comment]
Unless somebody has a better idea, I would suggest splitting this into two documents: One a general introduction to ZFS as related to the use in NASes, including the ability to move pools from FreeNAS to Ansible-NAS (and, of course, the reverse way); the other specific configuration suggestions and examples for the apps included.

**Comment by davestephens** [comment]
I think we need to consider the target audience too. A lot of people likely
won't care about most of the ZFS options, and just want to know that their
data is mirrored on a pair of disks and will be recoverable. I for one fall
into that camp.

Me aside, there are always power users that want to go "full enterprise" on
their home stack, and that's cool - but I've always been pretty open about
Ansible-NAS *not* managing disks/partitions because a- it's hideously
complicated and b- it's near-on impossible to automate for every use case.

I also don't believe there should be differing ZFS partition configs for
different applications - for the same reasons above, and also that not
everyone uses ZFS.

Any ZFS stuff should be set system-wide and used by everything - let's not
forget this is a home NAS, not an enterprise grade storage system where
disk latency of utmost importance :-)

Guiding new users through creating a sensible ZFS setup, and giving power
users ideas of options to tweak is absolutely the right thing to do, but I
think the documentation needs to call those differing use cases out
clearly.

Tl;Dr; I think we're agreeing, I just want to be clear that none of this
stuff is ever going to be automated, and I don't want to be recommending
loads of manual tweaking every time a new container is spun up.

**Comment by scotws** [comment]
Right, I see it exactly the same way - automating ZFS is asking for trouble, and a complete guide would be in book length and require constant updating. My idea would be:

A **"ZFS overview"** for total newbies: Move the stuff about ZFS out of the 'overview.md' and stick it in this document; explain advantages of ZFS (self-healing, don't care about ransomware with snapshots, can move pools to and from FreeNAS, Linux, FreeBSD at will, etc) and disadvantages, references to further help. -- Write this once and never touch it again until some change to the core ZFS structure (in other words, probably never).

A **"ZFS specific suggestions"** text with example command lines. My current take would be: We don't do your ZFS setup for you, but we'll give you some pointers here because we're nice that way. Specifically: Root on ZFS is still too complicated with Ubuntu, use ext4 (mirrored with mdadm if you want to); this means Docker stuff stays there as well. Example of two hard drives in mirrored configuration for ZFS in one single pool for all data; set scrub cron job for this pool. Set up datasets for movie and TV files with larger bock sizes; set up datasets for shares with "normal" block sizes; example snapshot configuration for these shares. Never ever use deduplication, and you'll probably not benefit at all from L2ARC or SLOG in our case, so just don't. -- We walk the user through a simple default sample configuration, and update this for relevant major new ZFS features (at rest encryption, easy root on ZFS in Ubuntu setup menu), but that's it.

I mean, if this is too long, I won't get to watch the stuff I'm trying to store in the first place 😄.
